src/app/modules/core/advertisement/advertisement.service.ts

```diff
@@ -0,0 +1,80 @@
+import { Injectable } from '@angular/core'
+import {
+  IMockAdvertisement,
+  MOCKADVERTISEMENTS_MOCK_DATA,
+} from '@modules/core/advertisement/mock-advertisement/mock-advertisement'
+import { IFacilities } from '@shared/models/property'
+import * as _ from 'lodash'
+import { Observable, of } from 'rxjs'
+import { map } from 'rxjs/operators'
+
+@Injectable({
+  providedIn: 'root',
+})
+export class AdvertisementService {
+  constructor() {}
+
+  findAll(): Observable<IMockAdvertisement[]> {
+    return of(MOCKADVERTISEMENTS_MOCK_DATA)
+  }
+
+  findById(id: number): Observable<IMockAdvertisement> {
+    return of(MOCKADVERTISEMENTS_MOCK_DATA.find((advert) => advert.id === id))
+  }
+
+  returnPriceFilteredAdvertisements(filter: number): Observable<IMockAdvertisement[]> {
+    return of(MOCKADVERTISEMENTS_MOCK_DATA).pipe(
+      map((filteredAdvertisements: IMockAdvertisement[]) => {
+        return filteredAdvertisements.filter((x) => x.price <= filter)
+      })
+    )
+  }
+
+  returnScoreFilteredAdvertisements(filter: number): Observable<IMockAdvertisement[]> {
+    const filteredArray: IMockAdvertisement[] = []
+    return of(MOCKADVERTISEMENTS_MOCK_DATA).pipe(
+      map((sourceArray: IMockAdvertisement[]) => {
+        sourceArray.forEach((advertisement) => {
+          const lenght = advertisement.reviews.length
+          const score = advertisement.reviews
+            .map((review) => review.vote)
+            .reduce((a, b) => Math.round((a + b) / lenght))
+          if (score >= filter) {
+            filteredArray.push(advertisement)
+          }
+        })
+        return filteredArray
+      })
+    )
+  }
+
+  returnFacilitiesFilteredAdvertisements(
```

Comment:
Non mi e' chiaro cosa tu stia cercando di fare con questa logica :-/

Reply:
@pc-codingavision Il metodo mi restituisce un array di advertisements in base al filtro di facilities che gli passo, esattamente quale passaggio non è chiaro? 😄

Reply:
La logica, secondo me, e' un po confusionaria. Ne parliamo a lezione.